docs/docs/guides/sign-in-with-github-google-facebook-linkedin.mdx

```diff
@@ -551,7 +552,90 @@ selfservice:
 Next, open the login endpoint of the SecureApp and you should see the Twitch
 Login option!
 
-## Google, LinkedIn, Facebook
+## Facebook
+
+### Create Facebook Application
+
+To set up "Sign in with Facebook" you must create a
+[Facebook OAuth2 Application](https://developers.facebook.com/apps/).
+
+Then add a "Facebook Login" product to this application.
+
+Set the "Valid OAuth Redirect URIs" to:
+
+```
+https://127.0.0.1:4433/self-service/methods/oidc/callback/facebook
+```
+
+The pattern of this URL is:
+
+```
+https://<domain-of-ory-kratos>:<public-port>/self-service/methods/oidc/callback/<provider-id>
+```
+
+Facebook requires https the redirect URIs!
+
+:::note
+
+While you can use
+[Facebook as an OIDC identity provider](https://developers.facebook.com/docs/facebook-login/),
+Facebook only returns access_token and not id_token. Therefore, ORY Kratos makes
+a request to
+[Facebook's graph.facebook.com/me API](https://graph.facebook.com/me) and adds
+the user info to `std.extVar('claims')`.
+
+:::
+
+### Configuring Kratos
+
+Create a Jsonnet claims mapper as described in
+[OpenID Connect and OAuth2 Credentials](../concepts/credentials/openid-connect-oidc-oauth2.mdx).
+Save the code in
+`<kratos-directory>/contrib/quickstart/kratos/email-password/oidc.facebook.jsonnet`.
+
+```json title="contrib/quickstart/kratos/email-password/oidc.facebook.jsonnet"
+local claims = std.extVar('claims');
+{
+  identity: {
+    traits: {
+      // Allowing unverified email addresses enables account
+      // enumeration attacks, especially if the value is used for
+      // e.g. verification or as a password login identifier.
+      //
+      // It is assumed that Facebook requires a email to be verifed before accessable via Oauth (because they don't provide an email_verified field).
+      //
+      // The email might be empty if the user is not allowed to an email scope.
+      [if "email" in claims then "email" else null]: claims.email,
+    },
+  },
+}
+```
+
+Now, enable the Facebook provider in the ORY Kratos config located at
+`<kratos-directory>/contrib/quickstart/kratos/email-password/kratos.yml`.
+
+```yaml title="contrib/quickstart/kratos/email-password/kratos.yml"
+# $ kratos -c path/to/my/kratos/config.yml serve
+selfservice:
+  methods:
+    oidc:
+      enabled: true
+      config:
+        providers:
+          - id: facebook # this is `<provider-id>` in the Authorization callback URL. DO NOT CHANGE IT ONCE SET!
+            provider: facebook
```

Comment:
```suggestion
            provider: generic
```
It only worked for me when I put `provider: generic` here. 
Did I do something wrong and it works with `provider: facebook` for you?

Reply:
Thank you @vinckr for the review! You have to compile Ory Kratos to get the code this PR implements, which enables the `facebook` provider. The error you saw is because this line was changed!

You need to checkout this fork/branch, then cd into the project directory, make the changes suggested in the docs to, and then https://github.com/ory/kratos/tree/master/contrib/quickstart/kratos/email-password and then compile the quickstart using the changes in this branch:

```
make quickstart-dev
```

This will start the Ory Kratos quickstart with the code changes included in this PR.

Let me know how it goes!

Reply:
Yes, it worked [that way](https://github.com/ory/kratos/pull/1035#discussion_r570780330) for me.

Reply:
Thanks for the tip, it worked now! 
I get redirected on facebook and get an error there, but that is because I dont have a proper setup. 

So the part from Kratos side works well, great work qvamatal 👍

Reply:
Should we add the compile step to the tutorial as well or some pointer to run make _after_ you make config changes? 
Or do we not need to compile after this is merged?

Reply:
The next release includes these code changes, so this only needs to be done when reviewing PRs or unreleased changes.